lxd/devices.go

```diff
@@ -366,53 +366,28 @@ func fillFixedInstances(fixedInstances map[int64][]instance.Instance, inst insta
 	}
 }
 
-// deviceTaskBalance is used to balance the CPU load across instances running on a host.
-// It first checks if CGroup support is available and returns if it isn't.
-// It then retrieves the effective CPU list (the CPUs that are guaranteed to be online) and isolates any isolated CPUs.
-// After that, it loads all instances running on the node and iterates through them.
-//
-// For each instance, it checks its CPU limits and determines whether it is pinned to specific CPUs or can use the load-balancing mechanism.
-// If it is pinned, the function adds it to the fixedInstances map with the CPU numbers it is pinned to.
-// If not, the instance will be included in the load-balancing calculation,
-// and the number of CPUs it can use is determined by taking the minimum of its assigned CPUs and the available CPUs. Note that if
-// NUMA placement is enabled (`limits.cpu.nodes` is not empty), we apply a similar load-balancing logic to the `fixedInstances` map
-// with a constraint being the number of vCPUs and the CPU pool being the CPUs pinned to a set of NUMA nodes.
-//
-// Next, the function balance the CPU usage by iterating over all the CPUs and dividing the instances into those that
-// are pinned to a specific CPU and those that are load-balanced. For the pinned instances,
-// it adds them to the pinning map with the CPU number it's pinned to.
-// For the load-balanced instances, it sorts the available CPUs based on their usage count and assigns them to instances
-// in ascending order until the required number of CPUs have been assigned.
-// Finally, the pinning map is used to set the new CPU pinning for each instance, updating it to the new balanced state.
-//
-// Overall, this function ensures that the CPU resources of the host are utilized effectively amongst all the instances running on it.
-func deviceTaskBalance(s *state.State) {
-	// Don't bother running when CGroup support isn't there
-	if !s.OS.CGInfo.Supports(cgroup.CPUSet, nil) {
-		return
-	}
-
+func getCPULists() (effectiveCpus string, cpus []int64, err error) {
```

Comment:
please can we have a comment on this function explaining what the return values are and their format, primarily the effectiveCpus string.

Also should this be called `effectiveCPUs`?

Reply:
Thanks! Fixed!